### PR TITLE
fixes for #860

### PR DESCRIPTION
**Pull request by sarveshwar-gavhane**
Fixes https://github.com/adaptlearning/adapt_framework/issues/860

### CODE REVIEW

**Comment by moloko** [comment]
original issue for others' reference [#860](https://github.com/adaptlearning/adapt_framework/issues/860)

**Comment by moloko** [comment]
+1

**Comment by oliverfoster** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

File: js/adapt-contrib-narrative.js [hunk 1]
```diff
@@ -182,6 +182,9 @@ define(function(require) {
 
         setStage: function(stage, initial) {
             this.model.set('_stage', stage);
+            if(this.model.get('_stage') === 0 && this.model.get('_items').length === 1) {
```
Comment (hrajotia) [inline js/adapt-contrib-narrative.js]: There is no need to check `_stage` here as we are only doing this for a single item.

I also think that these line should go to the [setupEventListeners](https://github.com/adaptlearning/adapt-contrib-narrative/blob/master/js/adapt-contrib-narrative.js#L353) method.

e.g.:
```javascript
            if (this.completionEvent !== 'inview' && this.model.get('_items').length > 1) {
                this.on(this.completionEvent, _.bind(this.onCompletion, this));
            } else {
                this.$('.component-widget').on('inview', _.bind(this.inview, this));
            }
```

**Comment by hrajotia** [comment]
+1

**Comment by oliverfoster** [comment]
yea, that makes more sense +1

**Comment by tomgreenfield** [comment]
+1

**Pull request merged**